source/Server.Contracts/Variables/ActionSpecialVariable.cs

```diff
@@ -0,0 +1,35 @@
+using System.Collections.Generic;
+
+namespace Sashimi.Server.Contracts.Variables
+{
+    public class ActionSpecialVariable
+    {
+        public ActionSpecialVariable(string name, BindingBehaviour userVisible = BindingBehaviour.Visible, params string[] deprecatedAliases)
+        {
+            Name = name;
+            DeprecatedAliases = deprecatedAliases;
```

Comment:
Why do we need this `DeprecatedAliases` property?

Reply:
There are a number of variables in server that have been renamed over time. The deprecated aliases contain the list of previous names, and during the deployment pipeline server will duplicate the current variable value into all of the old names as well, in case any existing scripts are using them. I.e. is for backward compat.

Reply:
hmm ... are we planning to/can we address this backward compact issue? If this can never be addressed somehow, maybe it should be considered as a feature here to support `Aliases` rather than a `DeprecatedAliases` as we usually use attribute-based approach for deprecation/obsolete instead of polluting the object itself to avoid any unnecessary API change of the object.